seatunnel-connectors-v2/connector-paimon/src/main/java/org/apache/seatunnel/connectors/seatunnel/paimon/sink/PaimonSinkWriter.java

```diff
@@ -182,7 +185,7 @@ public PaimonSinkWriter(
                             .flatMap(List::stream)
                             .collect(Collectors.toList());
             log.info("Trying to recommit states {}", commitables);
-            if (JobContextUtil.isBatchJob(jobContext)) {
+            if (JobContextUtil.isCheckpointNotEnabledInBatchMode(jobContext, envConfig)) {
```

Comment:
why we should care checkpoint enabled when restore? If checkpoint not enable, the restore will without any `commitables`.

Reply:
It just be used should we use the commit of  stream api or not. The stream api  can pass the checkpointId as an argument.

Reply:
Also, this code is inherited from history, I just added a condition. I don't think we should commit here. I'll do a separate pr and just restore the state without committing.

Reply:
done